tests/d3d12_enhanced_barriers.c

```diff
@@ -259,6 +259,260 @@ void test_enhanced_barrier_castable_formats_validation(void)
     ID3D12Device12_Release(device12);
 }
 
+void test_enhanced_barrier_castable_dsv(void)
+{
+    enum { PSO_INDEX_FLOAT = 0, PSO_INDEX_UINT, PSO_INDEX_UINT16, PSO_INDEX_COUNT };
+    D3D12_FEATURE_DATA_D3D12_OPTIONS12 features12;
+    ID3D12PipelineState *psos[PSO_INDEX_COUNT];
+    D3D12_UNORDERED_ACCESS_VIEW_DESC uav_desc;
+    D3D12_SHADER_RESOURCE_VIEW_DESC srv_desc;
+    D3D12_DEPTH_STENCIL_VIEW_DESC dsv_desc;
+    struct test_context_desc context_desc;
+    D3D12_DESCRIPTOR_RANGE desc_range[2];
+    ID3D12DescriptorHeap *resource_heap;
+    D3D12_CPU_DESCRIPTOR_HANDLE cpu_h;
+    D3D12_ROOT_SIGNATURE_DESC rs_desc;
+    D3D12_ROOT_PARAMETER rs_params[1];
+    D3D12_HEAP_PROPERTIES heap_props;
+    ID3D12DescriptorHeap *dsv_heap;
+    struct test_context context;
+    D3D12_RESOURCE_DESC1 desc1;
+    ID3D12Device10 *device10;
+    ID3D12Resource *resource;
+    ID3D12Resource *readback;
+    unsigned int i;
+    HRESULT hr;
+
+    static const struct test
+    {
+        DXGI_FORMAT tex_format;
+        DXGI_FORMAT dsv_format;
+        DXGI_FORMAT srv_format;
+        float clear_value;
+        uint32_t reference_value;
+        unsigned int pso_index;
+    } tests[] = {
+        { DXGI_FORMAT_D32_FLOAT, DXGI_FORMAT_D32_FLOAT, DXGI_FORMAT_R32_FLOAT, 1.0f, 0x3f800000, PSO_INDEX_FLOAT },
+        { DXGI_FORMAT_D32_FLOAT, DXGI_FORMAT_D32_FLOAT, DXGI_FORMAT_R32_FLOAT, 0.5f, 0x3f000000, PSO_INDEX_FLOAT },
+        { DXGI_FORMAT_D32_FLOAT, DXGI_FORMAT_D32_FLOAT, DXGI_FORMAT_R32_FLOAT, 0.0f, 0, PSO_INDEX_FLOAT },
```

Comment:
Should probably add a test for R32_FLOAT as base format with D32_FLOAT as castable. That might break our current impl ...